# Python으로 Excel 대체하기
# Contents
### 1. 출력
### 2. 입력
### 3. 조건
### 4. 반복

# 1. 출력

출력문은 Terminal 화면에 표시할 수 있게 하는 함수입니다. 다양한 함수들이 있지만, 우리가 편히 그리고 대부분 쓰는 함수는 print()라는 함수입니다. print()함수를 쓰는 법은 다양하게 있습니다. 아래의 예시들을 참고하여 익히시면 됩니다.




In [ ]:
print("Hello World")
print('Hello World')

Hello World
Hello World


print()를 사용할 때 가장 대표적인 방법은 " "(쌍따움표) 또는 ' '(따움표)를 사용하여 띄우는 방식입니다. 이때에는 따움표 안에 들어가있는 문장 또는 숫자가 그대로 출력이 됩니다. 또한 print()를 연속으로 사용하게 되면, 다음 print()함수 안에 들어가 있는 내용들은 자동으로 다음줄에 표현됩니다.

In [ ]:
a="Hello World"
print(a)

Hello World


다음은 변수에 저장된 내용을 출력하는 방법 입니다. 위의 코드에서는 a라는 변수에 Hello Wolrd 라는 값을 저장 시켰습니다. 여기서 a라는 값을 출력하고 싶으면, print()안에 a값을 넣으면 됩니다. 하지만, 변수의 값을 지정하기 위해서는 따음표나 쌍따음표가 있으면 안 됩니다.

In [ ]:
a=1
b=2
print(a+b)

3


위의 코드는 a에 1을, b에 2를 저장시킨 후, print()내부에서 a와 b의 합을 출력하는 코드입니다. 이를 통해 print()내부에서도 연산이 가능하다는 점을 알 수 있습니다.

In [ ]:
a=1
b=2
c=a+b
print(c)

3


위의 코드는 a값에 1을, b값에 2를 c값에 a+b라는 함수를 저장시킨 후, c값을 출력하는 것을 보여줍니다. 이를 통해 알 수 있는 것은 변수에 함수 혹은 수식을 저장할 수 있다는 점입니다.

In [ ]:
a=1
print(a)
a=3
print(a)
a=a+1
print(a)

1
3
4


위의 코드에서 확인할 수 있듯이 변수의 값은 덮어쓰는 것이 가능합니다. 1번째 줄에서 a에 1을 저장하였을 때는 a값을 출력하였을 때, 1이 출력되는 것을 확인할 수 있지만, 그 후에 a에 3을 다시 저장하였을 때는 3이 나오는 것을 확인 할 수 있습니다. 마지막 줄 처럼, a값에 자기자시의 값에 더하기 1을 한 값을 다시 저장할 수 있습니다. 그리고 3+1의 값인 4가 출력되는 것을 확인할 수 있습니다.

In [ ]:
'''
# 주석은 코드연산에 관여하지 않습니다.
'''

a=2
# a=1
print(a)


2


출력문에서 마지막으로 배울 것은 주석입니다. 주석은 코드연산에 아무 관여를 하지 않으며, 코드를 설명할 때 사용합니다. 주석을 한줄만 할 때는 #을 사용하며, 여러 줄을 주석으로 만들고 싶을 때는 위와 같이 ''' ''' 사이에 작성합니다.

# 2. 입력

입력문은 코드 외의 인터페이스에서 값을 입력받을 수 있게 하는 함수입니다. 다양한 함수들이 있지만, 우리가 편히 그리고 대부분 쓰는 함수는 input()라는 함수입니다. input()함수의 특징을 알아보겠습니다.



In [ ]:
a=input()
print(a)

mdfsaklmfmkasmfk
mdfsaklmfmkasmfk


In [ ]:
a=input("값을 입력해주세요")
print(a)

값을 입력해주세요123abc
123abc


값을 받을 때는 a라는 변수에 input() 값을 저장시켜주는 방식으로 진행됩니다. 만약에 값을 입력받을 때, 어떤 글을 띄어주고 싶다면, input()안 " "안에 적어주면 됩니다.

In [ ]:
a=input()
print(type(a))

12
<class 'str'>


In [ ]:
a=input()
b=input()
print(a+b)

1
2
12


input()을 통해 값을 받게 되면 값은 "str"이라는 데이터형태로 저장되는데, 이는 문자열이라는 뜻입니다. 문자열로 데이터를 받게 되면 연산이 불가능합니다. 그래서 값이 단순히 "1"과 "2"가 합쳐진 형태인 "12"로 나오는 것입니다. 연산을 위해서는 "int"라는 정수값이나 "float"라는 실수값으로 전환해주어야 합니다. 이렇게 해줄때는 아래와 같은 함수들을 사용합니다.

```
int(input())
```

```
float(input())
```

In [ ]:
a=int(input())
b=int(input())

c=float(input())
d=float(input())
print(a+b)
print(c+d)

1
2
1
2
3
3.0


int()는 정수값을 의미하기 때문에 소수점이 보이지 않지만, float()는 실수 값을 의미하기 때문에 소수점이 있다는 것을 확인할 수 있습니다.

In [ ]:
a=3.99
print(int(a))

3


만약에 입력받은 값은 실수인데 int()에 넣게 되면 소수점은 절삭해서 보여지는 것을 확인할 수 있습니다.

# 3. 조건

조건문은 프로그램에서 특정 조건이 참(True) 또는 거짓(False)일 때 다른 코드 블록을 실행하도록 하는 구문입니다. 파이썬에서 가장 일반적인 조건문은 if, else, elif 키워드를 사용한 것입니다. if는 가장 먼저 조건문을 선언할 때 사용되고, elif는 다른 if조건을 선언할 때 사용되며, else는 나머지 일 때 사용되는 함수입니다.

In [ ]:
a=int(input())
b=int(input())

if a>b:
  a=a+1
elif a==b:
  a=a+2
else:
  a=a+3

print(a)

3
4
6


위의 코드에서는 입력받은 a,b값을 통해 a값이 b값보다 작을 경우에는 a값에 1을 더하고, a값과 b값이 같을 경우에는 a에 2를 더하고, 마지막으로 그 외의 경우에는 a값에 3을 더하도록 하는 코드를 작성하였습니다.

# 4. 반복

반복문은 동일한 블록이나 일련의 작업을 여러 번 반복하는 데 사용됩니다. 파이썬에서 가장 일반적인 반복문은 for와 while입니다. 먼저 for문 같은 것은 원하는 만큼 반복할 때 사용되며, while의 경우에는 조건문과 합쳐진 반복문이라고 생각하면 됩니다.

In [ ]:
sum=0
for i in range(100):
  sum=sum+(i+1)
print(sum)

5050


먼저 for문을 살펴보겠습니다. 위의 코드는 1부터 100까지 더하는 코드입니다. 먼저 sum을 0으로 선언해주고, for i in range(100)을 하므로써 i가 0부터 99까지 반복한다라는 것을 적어주었습니다. 그 다음줄에서는 sum=sum+(i+1)을 하여, sum값에 전에 sum값에 순서대로 합산해줄 수 있습니다.

첫번째 반복해서는 i값은 0이고, sum값도 0입니다. 그래서 sum=0+(0+1)이기 때문에 처음 반복의 마지막 값은 sum=1이 될 것 입니다.

두번째 반복해서는 i값은 1이고 sum 값은 1이기 때문에 sum=1+(1+1)이 되어 2번째 반복 후의 값은 sum=3이 될 것입니다.

세번째 반복에서는 i값이 2이고 sum 값은 3이기 때문에 sum=3+(2+1)이 되어 sum값은 6이 될 것입니다.

이렇듯 i가 99가 될 때까지 반복하게 되면 1부터 100까지의 합인 5050이 나오게 될 것 입니다.

In [ ]:
a=0
while a<=4:
  print(a+1,"번째 반복입니다.")
  a=a+1

1 번째 반복입니다.
2 번째 반복입니다.
3 번째 반복입니다.
4 번째 반복입니다.
5 번째 반복입니다.


다음은 while문입니다. while문은 반복과 조건문이 합쳐진 형태입니다. 위의 코드는 a를 0으로 선언하고, a가 4와 같아지거나 작아질 때까지 반복하게 하는 코드입니다. a는 0부터 시작하기 때문에 a=1,2,3,4이렇게 올라갈 것이며, 5번 째 반복에서 a는 5가 되며 while 루프를 벗어나게 될 것 입니다.